rdmo/projects/handlers/project_save_tasks.py

```diff
@@ -0,0 +1,28 @@
+from django.db.models.signals import post_save, pre_save
+from django.dispatch import receiver
+
+from rdmo.projects.models import Project
+from rdmo.tasks.models import Task
+
+DEFERRED_SYNC_TASKS_KEY = '_deferred_sync_tasks'
+
+@receiver(pre_save, sender=Project)
+def pre_save_project_sync_tasks_from_catalog(sender, instance, raw, update_fields, **kwargs):
+    if raw or (update_fields and 'catalog' not in update_fields):
+        return
+
+    # Fetch the original catalog from the database
+    original_instance = sender.objects.get(id=instance.id)
+    if original_instance.catalog == instance.catalog:
+        # Do nothing if the catalog has not changed
+        return
+
+    # Defer synchronization of views
+    setattr(instance, DEFERRED_SYNC_TASKS_KEY, True)
```

Comment:
I needed a `pre_save` to check if the `catalog` was changed or not. And afterwards it passes this boolean (with deferred name) to the `post_save`.
It seems complicated for just a `project.save()` but could not find another way.

Reply:
remove setattr and just use, eg. `instance._catalog_has_changed_sync_task`